pcs/git_manager.py

```diff
@@ -204,39 +219,56 @@ def _get_default_repo_path(self, org_name: str, project_name: str) -> Path:
         global_default_path = AOS_GLOBAL_REPOS_DIR / "_default"
         default_repo_path = global_default_path / org_name / project_name
         url = f"https://github.com/{org_name}/{project_name}.git"
-        cloned = self._clone_repo(url=url, clone_destination=default_repo_path)
+        cloned = self._clone_repo(src=url, clone_destination=default_repo_path)
         if not cloned:
             self._maybe_fetch(default_repo_path)
         return default_repo_path
 
+    def _write_repo_info(self, repo_path: Path) -> dict:
+        default_info_path = self._get_repo_info_path(repo_path)
+        repo_info = {
+            self.REPO_INFO_PROC_KEY: os.getpid(),
+            self.REPO_INFO_TIME_KEY: time.time(),
+        }
+        with open(default_info_path, "w") as fout:
+            json.dump(repo_info, fout)
+        return repo_info
+
+    def _read_repo_info(self, repo_path: Path) -> Optional[dict]:
+        default_info_path = self._get_repo_info_path(repo_path)
+        with open(default_info_path) as fin:
+            try:
+                return json.load(fin)
+            except JSONDecodeError:
+                return None
+
+    def _get_repo_info_path(self, repo_path: Path) -> Path:
+        return Path(str(repo_path) + "-info.json")
+
     def _maybe_fetch(self, default_repo_path: Path) -> None:
         """
         This function periodically runs `git fetch` in the "default" repo used
         to map ref names to sha1 hashes.  We only update if we're a new
         process or if we haven't updated in the last half hour.
         """
-        PROC_KEY = "process_id"
-        TIME_KEY = "last_fetch_time"
-        default_info_path = Path(str(default_repo_path) + "-info.json")
 
         def _do_fetch():
-            with open(default_info_path, "w") as fout:
-                json.dump({PROC_KEY: os.getpid(), TIME_KEY: time.time()}, fout)
             print(f"GitManager: fetching {default_repo_path}...")
             with porcelain.open_repo_closing(default_repo_path) as repo:
                 porcelain.fetch(repo)
+            self._write_repo_info(default_repo_path)
 
-        if not default_info_path.exists():
+        info_path = self._get_repo_info_path(default_repo_path)
+        assert info_path.exists()
+        repo_info = self._read_repo_info(default_repo_path)
+        if repo_info is None:
             _do_fetch()
             return
-        with open(default_info_path) as fin:
-            try:
-                info = json.load(fin)
-            except JSONDecodeError:
-                _do_fetch()
-                return
-        sec_since_last_fetch = time.time() - info[TIME_KEY]
-        if info[PROC_KEY] != os.getpid() or sec_since_last_fetch > 1800:
+        if repo_info[self.REPO_INFO_PROC_KEY] != os.getpid():
+            _do_fetch()
+            return
+        sec_since_last_fetch = time.time() - repo_info[self.REPO_INFO_TIME_KEY]
+        if sec_since_last_fetch > 1800:
```

Comment:
Can this be an `elif` so that we don't do the fetch twice in a row when both conditions are true?

Reply:
I think this should be okay because we return immediately from `_maybe_fetch()` if we ever end up calling `_do_fetch()`  so no opportunity for a double fetch (it's hard to see in the diff view but clearer in the code, I think).